includes/Core/Remote_Features/Remote_Features_Syncer.php

```diff
@@ -78,6 +88,20 @@ public function pull_remote_features() {
 
 		if ( ! is_wp_error( $features ) && is_array( $features ) ) {
 			$this->remote_features->set( $features );
+			$this->remote_features_last_sync->set( time() );
```

Comment:
I'm not sure this is necessary. Instead of introducing an additional persistent feature-related option, why not simply store the last updated time whenever we call `set`? Maybe this would be better as a new method on `Remote_Features` like 

```php
public function update( $features ) {
    $features['last_updated_at'] = time();
    $this->set( $features );
}
```

And then this could be a simple one line change
```suggestion
			$this->remote_features->update( $features );
```